runtime-modules/proposals/codex/src/lib.rs

```diff
@@ -437,7 +457,11 @@ decl_module! {
             Self::ensure_details_checks(&proposal_details)?;
 
             let proposal_parameters = Self::get_proposal_parameters(&proposal_details);
-            let proposal_code = T::ProposalEncoder::encode_proposal(proposal_details);
+            // TODO: encode_proposal could take a reference instead of moving to prevent cloning
```

Comment:
Makes sense. Create an issue, please, and add it to the comment.

Reply:
Done #2161